pkg/server/ingest_test.go

```diff
@@ -328,12 +328,14 @@ var _ = Describe("server", func() {
 					for _, t := range types {
 						func(t string) {
 							type contextID struct {
-								id  string
-								key string
+								id    string
+								sleep time.Duration
+								key   string
 							}
 							cids := []contextID{
-								{id: "0", key: `test.app.` + t + `{foo="bar", baz="qux"}`},
-								{id: "1", key: `test.app.` + t + `{foo="bar", baz="qux", thread_name="pool-2-thread-8"}`},
+								{id: "0", sleep: 500 * time.Millisecond, key: `test.app.` + t + `{foo="bar", baz="qux"}`},
+								{id: "1", sleep: 500 * time.Millisecond, key: `test.app.` + t + `{foo="bar", baz="qux", thread_name="pool-2-thread-8"}`},
+								{id: "2", sleep: 6 * time.Second, key: `test.app.` + t + `{foo="bar", baz="qux", thread_name="pool-2-thread-8",profile_id="239239239239"}`},
```

Comment:
6 seconds seems a bit too much? Can we do a lower value?

Reply:
It is because of `exemplarBatchDuration = time.Second * 5`, we could either lower the `exemplarBatchDuration` for the test or add some force flush logic, I am not sure which is better or maybe there are other options, I'd ask @kolesnikovae opinion.
Also the other problem here is that we do this 8 times because of the nested contexts - ones per every type of sample(cpu, objects, ...etc). It would be nice to rewrite the test to put the jfr once and query the date 8 times. Should I do it in this PR?

Reply:
> It is because of exemplarBatchDuration = time.Second * 5, we could either lower the exemplarBatchDuration for the test or add some force flush logic

I'm afraid flushing exemplars from outside is only available in https://github.com/pyroscope-io/pyroscope/pull/1169: it adds `storage.ExemplarsInternals()` which also returns `Sync()` callback. I think I'll merge it soon, so you can rebase.

I'd say that tests in this package should not depend on the actual storage implementation. It's better to use an ingester mock. As of now our tests cover multiple "layers":
- ingestion - raw HTTP request -> `ingestion.IngestInput`
- parsing: `ingestion.IngestInput` -> `storage.PutInput`
- persistent storage: `storage.PutInput` -> data in BadgerDB

I won't nag about absolutely hermetic unit tests but we could do this better.
- If the ingester handler is only responsible for processing HTTP requests to `ingestion.IngestInput`, our test should validate only this conversion.
- Details of how raw profiling data of one of the format is parsed should be covered in the parser tests.

Reply:
I have just merged https://github.com/pyroscope-io/pyroscope/pull/1169 - sorry for the delay